(potentially) fix theme switch flash on websites using ThemeSwitch component

Yusyuriv opened the pull request:
The current implementation of the ThemeSwitch component has an issue: it waits for the `DOMContentLoaded` event before setting the theme saved in localStorage. This causes two issues:

* A bright flash for split second when the page is already partially displaying, but the event hasn't fired yet.
* The icon for the light theme is displayed for split second before being replaced with the dark theme icon.

This PR does two things:

1. It adds a small inline script that sets the theme saved in localStorage immediately instead of waiting for `DOMContentLoaded`.
2. Hides dark/light theme icons based on `html[data-theme]` instead of `.ac-theme-switch-wrapper:has([aria-checked='true'])`, as those are only set later, on `DOMContentLoaded`.

Presumably, this component is usually located at the very top of the page, so the inline script runs almost immediately, changing the theme, and preventing the background/icon flash.

Yusyuriv committed:
(potentially) fix theme switch flash on websites using ThemeSwitch component

denv17 commented:
Thank you for the contribution! Please maintain the use of arrow functions for consistency across the codebase. Additionally, all script logic should be integrated into the same <script> tag to keep the code cleaner and more organized.

Yusyuriv committed:
style: replace function statement with arrow function

Yusyuriv commented:
I replaced the function statement with an arrow function.

There's a little issue with keeping all logic in one tag: this one small piece of code **must** be inlined and not located in a separate JS file so it's executed by the browser as soon as possible. The other part of the code imports stuff from another file, so it needs to be processed by Astro and is fine to leave as is. This is why I added it in a separate script with `is:inline` attribute.

denv17 commented:
Thank you, it has already been published :)
